web3.md

```diff
@@ -69,14 +69,15 @@ The `blockList` cell stores a list of previous blocks and network states.
     rule <k> #pushBlockchainState => . ... </k>
          <blockList> (.List => ListItem({ <network> NETWORK </network> | <block> BLOCK </block> })) ... </blockList>
          <network> NETWORK </network>
+         <blockhashes> (.List => ListItem(#blockchainItemHash({ <network> NETWORK </network> | <block> BLOCK </block> }))) ... </blockhashes>
          <block>   BLOCK   </block>
 
     syntax BlockchainItem ::= #getBlockByNumber ( BlockIdentifier , List , BlockchainItem ) [function]
  // --------------------------------------------------------------------------------------------------
     rule #getBlockByNumber( _:Int            , .List                 , _     ) => .BlockchainItem
     rule #getBlockByNumber( LATEST           , .List                 , BLOCK ) => BLOCK
     rule #getBlockByNumber( LATEST           ,   ListItem( BLOCK ) _ , _     ) => BLOCK
-    rule #getBlockByNumber( PENDING          , _                     , BLOCK ) => BLOCK
+    rule #getBlockByNumber( PENDING => LATEST, _                     , _     )
     rule #getBlockByNumber( EARLIEST         , .List                 , BLOCK ) => BLOCK
```

Comment:
I don't think we should be treating `PENDING` and `LATEST` as the same cases, it seems like these should be treated differently because otherwise why would there be two separate identifiers for it in other clients?

Reply:
I'm just mimicking ganache here. Ganache literally does a check on the blocknumber like this:
```js
if (number === "latest" || number === "pending") {
    return this.getHeight(callback);
}
```
If we want to change this to make more sense and actually follow the web3 rpc spec, we'll also have to change some of our tests since they expect ganache's behavior.

Reply:
Can you link the code where Ganache uses this?

Reply:
https://github.com/trufflesuite/ganache-core/blob/25de0e2348c0e49dd6a30f69e3a6889a2cca8048/lib/blockchain_double.js#L228